enhancements/api-review/custom-tags-aws.md

```diff
@@ -79,65 +101,373 @@ The userTags field is intended to be set at install time and is considered immut
+
+2) There is no tag with same key present, new tag is created for the AWS resource. In case of limit reached, a validation error is generated.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value_comp1`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`
+
+   Action = A new tag is created for AWS resource.
+
+   Final tag set to AWS resource = `key_infra1 = value_infra1`
+
+   Event action = An event is generated to notify user about the action status (success/failure) to create tags for the AWS resource.
+
+#### Update tags scenarios
+Users can update the user-defined tags by editing `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type.
+On update of `resourceTags`, AWS resource is not created or restarted.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already a tag with the same key and value present for AWS resource, the tag for the AWS resource will not be updated.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value_update1`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`
+
+   Action = There is no update for AWS resource.
+
+   Final tag set to AWS resource = `key_infra1 = value_update1`
+
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) There is already a tag with the same key and different value present for AWS resource, the user-define tag value for the AWS resource will be updated.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value_old`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`
+
+   Action = Existing tag for AWS resource is updated to reflect new value.
+
+   Final tag set to AWS resource = `key_infra1 = value_update1`
+
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+#### Delete tags scenarios
+Tags are deleted when the user sets the user-defined tag value to an empty string in `.spec.platformSpec.aws.resourceTags`.
+Also refer to `Precedence` scenario to understand cases where deletion of user-defined tags is not allowed .
+
+For example,
+
+Existing tag for AWS resource = `key_infra1 = value_old`
+
+New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`
+
+Action = Existing tag for AWS resource is deleted.
+
+Final tag set to AWS resource = deleted
+
+Event action = An event is generated to notify user about the action status (success/failure) to delete tags for the AWS resource.
+
+#### Precedence scenarios
+1) User-defined tags on kube resources MUST continue to take precedence during create and update. User-defined tags found on kube resources must not be deleted by methods described in `Delete tags scenarios`.
```

Comment:
What are "user-defined tags on kube resources"? Is that the same as "user-defined tags in an individual component"? If so, let's be consistent with terminology.

Reply:
Yes, agreed. Updated all to use "kube resource" terminology.